dvc/updater.py

```diff
@@ -40,36 +38,87 @@ def check(self):
-        c_major, c_minor, c_patch = [int(x) for x in current.split('.')]
+        if self._is_outdated():
+            self._notify()
 
-        if l_major <= c_major and \
-           l_minor <= c_minor and \
-           l_patch <= c_patch:
-            return
+    def _get_latest_version(self):
+        r = requests.get(self.URL, timeout=self.TIMEOUT_GET)
+        j = r.json()
+        self.latest = j['version']
+        open(self.updater_file, 'w+').close()
+
+    def _is_outdated(self):
+        l_major, l_minor, l_patch = [int(x) for x in self.latest.split('.')]
+        c_major, c_minor, c_patch = [int(x) for x in self.current.split('.')]
 
-        msg = (
+        return (l_major > c_major or
+                l_minor > c_minor or
+                l_patch > c_patch)
+
+    def _notify(self):
+        message = (
             'Update available {red}{current}{reset} -> {green}{latest}{reset}'
-            '\n'
-            'Run {yellow}pip{reset} install dvc {blue}--upgrade{reset}'
-            .format(
-                red=colorama.Fore.RED,
-                reset=colorama.Fore.RESET,
-                green=colorama.Fore.GREEN,
-                yellow=colorama.Fore.YELLOW,
-                blue=colorama.Fore.BLUE,
-                current=current,
-                latest=latest,
-            )
-        )
-
-        Logger.box(msg, border_color='yellow')
+            + '\n'
+            + self._get_update_instructions()
+        ).format(red=colorama.Fore.RED,
+                 reset=colorama.Fore.RESET,
+                 green=colorama.Fore.GREEN,
+                 yellow=colorama.Fore.YELLOW,
+                 blue=colorama.Fore.BLUE,
+                 current=self.current,
+                 latest=self.latest)
+
+        if sys.stdout.isatty():
+            Logger.box(message, border_color='yellow')
+
+    def _get_update_instructions(self):
+        instructions = {
+            'pip': 'Run {yellow}pip{reset} install dvc {blue}--upgrade{reset}',
+            'yum': 'Run {yellow}yum{reset} update {blue}dvc{reset}',
+            'yay': 'Run {yellow}yay{reset} {blue}-S{reset} dvc',
+            'brew': 'Run {yellow}brew{reset} upgrade {blue}dvc{reset}',
+            'apt': ('Run {yellow}apt-get{reset} install'
+                    ' {blue}--only-upgrade{reset} dvc'),
+            None: ('To upgrade follow this steps:\n'
+                   '1. Uninstall dvc binary\n'
+                   '2. Go to {blue}https://dvc.org{reset}\n'
+                   '3. Download and install new binary'),
+        }
+
+        package_manager = self._get_package_manager()
+
+        return instructions[package_manager]
+
+    def _get_package_manager(self):
+        package_managers = {
+            'rhel': 'yum',
+            'centos': 'yum',
+            'fedora': 'yum',
+            'amazon': 'yum',
+            'opensuse': 'yum',
+            'arch': 'yay',
+            'ubuntu': 'apt',
+            'debian': 'apt',
+            'darwin': 'brew',
```

Comment:
On darwin we have three options to install - brew, native pkg, pip. Should we check it the same way we do for pip? Make sure also that this check doesn't run too long (a second or two are fine?) Otherwise, we need to run it asynchronously I would say in a separate process.

Reply:
Indeed, @shcheklein , I think we should run the whole notifier async, as it takes quite some time retrieving the latest version from the API. Maybe we can run the whole thing in a forked process and write the version and installation method on `.dvc/updater`, that way, we can notify the user on every command without _phoning home_ or inspecting the _user land_.

I'll open a new issue for that

Reply:
For native pkg, @shcheklein , is there an standard path to store the installed package that we can search for? something like `/usr/lib/python*/site-packages/dvc`.

Also, I'm not sure if the `brew` message is correct, as we only provide a custom formula, could you correct it if so?

Thanks for the comments, they were great :+1:

Reply:
Yes, it's a custom formula. Still it will look strange if show `pip` message for brew installed package. I'm not sure about the standard path.

Reply:
> Indeed, @shcheklein , I think we should run the whole notifier async, as it takes quite some time retrieving the latest version from the API. Maybe we can run the whole thing in a forked process and write the version and installation method on `.dvc/updater`, that way, we can notify the user on every command without _phoning home_ or inspecting the _user land_.
> 
> I'll open a new issue for that

My concern was more about running `pip show` and potentially `brew list` - they take long time to run sometimes. Running API call asynchronously makes total sense though and it's a great idea by itself.